reframe/core/schedulers/slurm.py

```diff
@@ -266,6 +266,12 @@ def _check_and_cancel(self, reason_descr):
             # no reason details
             reason, reason_details = reason_descr, None
 
+        # Here we handle the case were the UnavailableNodes list is empty,
+        # which actually means that the job is pending
+        if reason == 'ReqNodeNotAvail' and reason_details:
+            if not re.match(r'UnavailableNodes:\S+', reason_details):
```

Comment:
I think this regex is a bit restrictive, cos you require that whatever comes after `:` has no whitespaces. I'd use instead the following:

```python
if re.match(r'UnavailableNodes:$', reason_details):
    return
```

Reply:
This whole `if` should go inside the next one, which checks if the reason is of a cancellation type.

Reply:
Another thing to remove is comment about "inaccurate SLURM documentation" in the definition of the `_cancel_reasons` list.